src/helper.cc

```diff
@@ -885,6 +889,21 @@ helper::handleFewerServers(const bool madeProgress)
         else
             fatalf("The %s helpers are crashing too rapidly, need help!", id_name);
```

Comment:
> PR description: Existing handleFewerServers() already handles this situation by killing Squid, but that code effectively excludes several use cases (e.g., it ignores common startup=0 helper configurations).

The above PR description text refers to `if (n_active < childs.n_startup && ...)` guard  located above this change request. I think the text is correct, _but_ I am worried that there is another guard that this PR is _not_ addressing -- `!needNew`. AFAICT, that guard is also correct, but this new cleanup logic should apply before we return from this function, regardless of _where_ we return from this function.

If you agree, please refactor, probably by moving the code above inside the `if (needNew)` statement and removing that explicit `return`.

We could also change needNew() to never return zero when there are queued transactions, but I think that would be wrong: Based on Squid configuration, needNew() has already said "no new helpers are needed". We should not violate our configuration (and start new helpers) just because a helper failure now requires us to kill some queued transactions.

Reply:
> this PR is not addressing -- !needNew.

Should we report an error (or kill Squid) if there are enough active helper processes? We could probably report sth (at a level > 1), but it seems that there is no emergency in this case, as far as there is a helper that can still do its work (regardless of other helper crashes).